Sourcery/Parsing/Parser.swift

```diff
@@ -453,18 +456,57 @@ extension Parser {
 
         return annotations
     }
+    
+    fileprivate func parseTypealiases(from source: [String: SourceKitRepresentable], containingType: Type?, processed: [[String: SourceKitRepresentable]]) -> [Typealias] {
+        var contents = self.contents
+        
+        //replace all processed substructures with whitespaces so that we don't process their typealiases again
+        for substructure in processed {
+            if let offset = substructure[SwiftDocKey.offset.rawValue] as? Int64,
```

Comment:
I'm not quite sure that this will work as `replacingCharacters` needs `Range<Self.Index>` and `SubstringIdentifier.key.range(for: source)` still returns ints 🤔 We might though change it to `replacingOccurrences` but I think it's not better from performance side (we already have a range, why to search for it again)

Maybe something like this:

```swift
if let range = SubstringIdentifier.key.range(for: source)
  .flatMap({ contents.byteRangeToNSRange(start: Int($0.offset), length: Int($0.length)) }) {
            
    let replacement = String(repeating: " ", count: range.length)
    contents = contents.bridge().replacingCharacters(in: range, with: replacement)
}
```